code/modules/atmospherics/machinery/pipes/multiz.dm

```diff
@@ -0,0 +1,53 @@
+/// This is an atmospherics pipe which can relay air up/down a deck.
+/obj/machinery/atmospherics/pipe/multiz
+	name = "multi deck pipe adapter"
+	desc = "An adapter which allows pipes to connect to other pipenets on different decks."
+	icon_state = "adapter-3"
+	icon = 'icons/obj/atmospherics/pipes/multiz.dmi'
+
+	dir = SOUTH
+	initialize_directions = SOUTH
+
+	layer = HIGH_OBJ_LAYER
+	device_type = UNARY
+	paintable = FALSE
+
+	construction_type = /obj/item/pipe/directional
+	pipe_state = "multiz"
+
+	var/mutable_appearance/center = null
+	var/mutable_appearance/pipe = null
+	var/obj/machinery/atmospherics/front_node = null
+
+/* We use New() instead of Initialize() because these values are used in update_icon()
+ * in the mapping subsystem init before Initialize() is called in the atoms subsystem init.
+ * This is true for the other manifolds (the 4 ways and the heat exchanges) too.
+ */
+/obj/machinery/atmospherics/pipe/multiz/New()
```

Comment:
you should use Initialize() instead of new and call parent (`. = ..()`) instead of `return ..()`

Reply:
Read the comment above.

```
/* We use New() instead of Initialize() because these values are used in update_icon()
 * in the mapping subsystem init before Initialize() is called in the atoms subsystem init.
 * This is true for the other manifolds (the 4 ways and the heat exchanges) too.
 */
```

`. = ..()` and `return ..()` shouldn't change anything here

Reply:
well ok then